xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/DefaultWikiObjectComponentManagerEventListener.java

```diff
@@ -87,11 +94,13 @@ public void onEvent(Event event, Object source, Object data)
             EntityReference xObject = xObjectEvent.getReference();
 
             // If the modified XObject can produce a WikiComponent
-            if (this.wikiObjectComponentManagerEventListenerProxy.getWikiObjectsList()
-                    .contains(((BaseObjectReference) xObject).getXClassReference())) {
+            WikiObjectComponentBuilder componentBuilder =
+                    this.getAssociatedComponentBuilder(((BaseObjectReference) xObject).getXClassReference());
```

Comment:
This cast might be a bit dangerous since there is nothing in the event specification about the reference being a BaseObjectReference, you might end up with an ObjectReference or even an EntityReference of type Object. I think I would get the actual object and then call getXClassReference() on it.